delta-api/src/main/java/io/cdap/delta/api/assessment/ColumnDetail.java

```diff
@@ -26,15 +28,13 @@
   private final String name;
   private final SQLType type;
   private final boolean nullable;
-  private final int length;
-  private final int scale;
+  private final Map<String, String> properties;
 
-  public ColumnDetail(String name, SQLType type, boolean nullable, int length, int scale) {
+  public ColumnDetail(String name, SQLType type, boolean nullable, @Nullable Map<String, String> properties) {
```

Comment:
In general it's better not to allow null collections and use empty collections instead.

Also, let's have 2 constructors, one that takes a Map and one that doesn't. Most types won't use properties, and this also won't break compilation of plugins.